base/compiler/methodtable.jl

```diff
@@ -94,12 +94,28 @@ is no least element. In both cases `nothing` is returned.
 function findsup(@nospecialize(sig::Type), table::InternalMethodTable)
     min_valid = RefValue{UInt}(typemin(UInt))
     max_valid = RefValue{UInt}(typemax(UInt))
-    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
-                   sig, table.world, min_valid, max_valid)::Union{MethodMatch, Nothing}
+    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
+                   sig, nothing, table.world, min_valid, max_valid)::Union{MethodMatch, Nothing}
     result === nothing && return nothing
     (result.method, WorldRange(min_valid[], max_valid[]))
 end
 
+function findsup(@nospecialize(sig::Type), table::OverlayMethodTable)
+    min_valid = RefValue{UInt}(typemin(UInt))
+    max_valid = RefValue{UInt}(typemax(UInt))
+    result = ccall(:jl_gf_invoke_lookup_worlds, Any, (Any, Any, UInt, Ptr{Csize_t}, Ptr{Csize_t}),
+                   sig, table.mt, table.world, min_valid, max_valid)::Union{MethodMatch, Nothing}
+    if result === nothing
+        # fall back to the internal method table
+        min_valid[] = typemin(UInt)
+        max_valid[] = typemax(UInt)
```

Comment:
technically not legal to reset this, but is perfectly acceptable not to
```suggestion
```

Reply:
I decided to make a duplicated call to `findsup` here.

Reply:
ah, makes sense

though note it is mandatory to preserve the lookup min_valid/max_valid regardless of whether you get a result

note 2 that it is also always mandatory to execute the "fall back to the internal method table" code, unless you can prove that the results are fully covered